pkg/controllers/termination/terminate.go

```diff
@@ -77,7 +81,15 @@ func (t *Terminator) drain(ctx context.Context, node *v1.Node) (bool, error) {
 		}
 		// Don't attempt to evict a pod that's already evicting
 		if !pod.DeletionTimestamp.IsZero() {
-			continue
+			// If the kubelet cannot register a pod as deleted, we need to force delete the pod in order to terminate the node
+			if time.Since(pod.DeletionTimestamp.Time) > time.Duration(*pod.DeletionGracePeriodSeconds) {
```

Comment:
To avoid NPEs, we should use something like `ptr.Int64Value(pod.DeletionGracePeriodSeconds)`